src/pages/costModels/costModel/index.tsx

```diff
@@ -47,7 +47,24 @@ class CostModelInformation extends React.Component<Props, State> {
     this.state = { tabIndex: 0 };
   }
 
+  public componentDidUpdate(prevProps, prevState) {
+    const { uuid } = this.props.match.params;
+    const { uuid: prevUuid } = prevProps.match.params;
+    const { tabIndex } = this.state;
+    const { tabIndex: prevTabIndex } = prevState;
+    if (prevTabIndex !== tabIndex || prevUuid !== uuid) {
+      insights.chrome.appObjectId(`uuid:${uuid}_tabIndex:${tabIndex}`);
+    }
+  }
+
+  public componentWillUnmount() {
+    insights.chrome.appObjectId(undefined);
+  }
+
   public componentDidMount() {
+    const { uuid } = this.props.match.params;
+    const { tabIndex } = this.state;
+    insights.chrome.appObjectId(`uuid:${uuid}_tabIndex:${tabIndex}`);
```

Comment:
Is identifying each table item with a UUID necessary? Do these particular UUIDs (e.g., a9ffd77a-b8ea-4107-8836-78ad618fb2a8) convey any useful / readable information?

Pendo will be notified when the cost models and/or cost models details pages are loader, but probably don't need to identify the individual table item. It's still the same details page, just viewed with different data.

I'm thinking that we could provide a simpler ID for each page? For example, 'cost-models', 'cost-models-details', etc. The tab index may be useful, but suspect the UUIDs are a bit much.

We could reach out to a designer and ask exactly what are they looking for from the cost models details page? However, designers don't typically ask for this level of detail in their "design share" meetings.

That said, the `componentDidMount` code could be located in a component wrapper (i.e., if we use our page paths as a generic `appObjectId`)?

Reply:
Um, good question. I think it can be useful but not necessary.

Reply:
Don't feel strongly about it, just curious.